### PR TITLE
Gra 1100 netmaker

### PR DESCRIPTION
**Pull request by mattkasun**
workflow updates
    

- add goreleaser.yml
      - use goreleaser action to build/publish netmaker/nmctl release assets
- streamline Dockerfile
- update go-builder to run on push to develop

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
-ENV GO111MODULE=auto
 
-RUN apk add git
-RUN GOOS=linux CGO_ENABLED=1 go build -ldflags="-s -X 'main.version=${version}'" -tags ${tags} .
+RUN GOOS=linux CGO_ENABLED=1 go build -ldflags="-s -w" -tags ${tags} .
 # RUN go build -tags=ee . -o netmaker main.go
 FROM alpine:3.16.2
 
-# add a c lib
-RUN apk add gcompat iptables wireguard-tools
```
Comment (0xdcarns) [inline Dockerfile:15]: gcompat was needed for sqlite, is this not the case now?

Comment (mattkasun) [reply Dockerfile:15]: no errors during docker image build and no unusual errors running container

**Review by 0xdcarns — APPROVED** [review]

**Comment by 0xdcarns** [comment]
@theguy951357 please test an image with this dockerfile

**Review by theguy951357 — APPROVED** [review]
image builds
server made from image runs ok and nodes have connectivity
pass

**Pull request merged**